BLE_GattServer_FOTAService/mock/source/main.cpp

```diff
@@ -0,0 +1,236 @@
+
+        _fota_service.set_event_handler(this);
+
+        start_advertising();
+    }
+
+    void start_advertising()
+    {
+        ble::AdvertisingParameters adv_parameters(
+                ble::advertising_type_t::CONNECTABLE_UNDIRECTED,
+                ble::adv_interval_t(ble::millisecond_t(100))
+        );
+
+        _adv_data_builder.setFlags();
+        _adv_data_builder.setAppearance(ble::adv_data_appearance_t::UNKNOWN);
+        _adv_data_builder.setName(DEVICE_NAME);
+
+        ble_error_t error = _ble.gap().setAdvertisingParameters(
+                ble::LEGACY_ADVERTISING_HANDLE,
+                adv_parameters
+        );
+
+        if (error) {
+            printf("_ble.gap().setAdvertisingParameters() failed\r\n");
+            return;
+        }
+
+        error = _ble.gap().setAdvertisingPayload(
+                ble::LEGACY_ADVERTISING_HANDLE,
+                _adv_data_builder.getAdvertisingData()
+        );
+
+        if (error) {
+            printf("_ble.gap().setAdvertisingPayload() failed\r\n");
+            return;
+        }
+
+        error = _ble.gap().startAdvertising(ble::LEGACY_ADVERTISING_HANDLE);
+
+        if (error) {
+            printf("_ble.gap().startAdvertising() failed\r\n");
+            return;
+        }
+
+        printf("Device advertising, please connect\r\n");
+    }
+
+    FOTAService::StatusCode_t on_binary_stream_written(FOTAService &fota_service,
+                                                       mbed::Span<const uint8_t> buffer) override
+    {
+        printf("bsc written: ");
+        for(uint8_t byte : buffer) {
+            printf("%02X", byte);
+        }
+        printf("\r\n");
+
+        return FOTAService::FOTA_STATUS_OK;
+    }
+
+    GattAuthCallbackReply_t on_control_written(FOTAService &fota_service,
+                                               mbed::Span<const uint8_t> buffer) override
+    {
+        auto opcode = static_cast<FOTAService::OpCode_t>(buffer[0]);
+
+        switch(opcode) {
+            case FOTAService::FOTA_NO_OP:
+            {
+                break;
+            }
+
+            case FOTAService::FOTA_START:
+            {
+                printf("Starting FOTA session\r\n");
+                fota_service.start_fota_session();
+                uint8_t status = FOTAService::FOTA_STATUS_OK;
+                fota_service.notify_status(mbed::make_const_Span(&status, 1));
+                break;
+            }
+
+            case FOTAService::FOTA_STOP:
+            {
+                printf("Stopping FOTA session\r\n");
+                fota_service.stop_fota_session();
+                break;
+            }
+
+            case FOTAService::FOTA_COMMIT:
+            {
+                printf("Committing the update\r\n");
+                disconnect(ble::local_disconnection_reason_t::USER_TERMINATION);
+                break;
+            }
+
+            default:
+            {
+                return static_cast<GattAuthCallbackReply_t>
```

Comment:
why are you casting a const value? can you not use the value you are casting into direclty?

Reply:
Can't implicitly cast `FOTAService::ApplicationError_t` to `GattAuthCallbackReply_t`.

Reply:
I see now: /* 0x80 - 0x9F => Application Error */